### DISCUSSION TITLE
Handling of labelling modifications in TheoreticalSpectrumGenerator

### DISCUSSION THREAD

**Discussion by jpfeuffer** [OpenMS/OpenMS#7233] [General]
@timosachsenberg If I am not mistaken, TSG currently shifts the Nterm fragments by the TMT tag, when you generate a spectrum for an ID with that modification. How should this be handled best? Doesn't the modification "fall off"/fragment most of the time to generate the reporter ions?
Would you say it's the library users task to remove the mod before subjecting it to TSG or should TSG handle that?

**Comment by timosachsenberg** [comment]
Hmm good point. two approaches: either put a lot of logic or special handling into TSG for different mods or keeping it conceptually simpler and let the user decide. I think letting the user decide is more practical and maybe handle this in the higher level class e.g. spectrumannotator .
